### ISSUE TITLE
45.3 After refit, cant open "Repair Bay" tab in MekHQ

### ISSUE THREAD

**Issue by Arakash0** [MegaMek/mekhq#1138]
### Environment
MeqHQ 0.45.3
Java version 1.8.0_181
Windows 7

### Description
On the day a refit of a Mech (an ON-1MA) was finished, when I pressed the repair bay tab the repair view did not appear. I believe the mech was showing as damaged in the hanger list.

After saving and reloading, the problem seemed to be fixed.

Following error is visible in the mekhqlog:
```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at mekhq.campaign.parts.equipment.AmmoBin.isOmniPodded(AmmoBin.java:812)
	at mekhq.campaign.parts.equipment.AmmoBin.getActualTime(AmmoBin.java:443)
	at mekhq.campaign.parts.Part.getTimeLeft(Part.java:826)
	at mekhq.campaign.parts.equipment.AmmoBin.getDesc(AmmoBin.java:497)
	at mekhq.gui.model.TaskTableModel.getValueAt(TaskTableModel.java:53)
	at javax.swing.table.TableRowSorter$TableRowSorterModelWrapper.getValueAt(Unknown Source)
	at javax.swing.DefaultRowSorter.compare(Unknown Source)
	at javax.swing.DefaultRowSorter.access$100(Unknown Source)
	at javax.swing.DefaultRowSorter$Row.compareTo(Unknown Source)
	at javax.swing.DefaultRowSorter$Row.compareTo(Unknown Source)
	at java.util.ComparableTimSort.countRunAndMakeAscending(Unknown Source)
	at java.util.ComparableTimSort.sort(Unknown Source)
	at java.util.Arrays.sort(Unknown Source)
	at javax.swing.DefaultRowSorter.sort(Unknown Source)
	at javax.swing.DefaultRowSorter.allRowsChanged(Unknown Source)
	at javax.swing.JTable.notifySorter(Unknown Source)
	at javax.swing.JTable.sortedTableChanged(Unknown Source)
	at javax.swing.JTable.tableChanged(Unknown Source)
	at javax.swing.table.AbstractTableModel.fireTableChanged(Unknown Source)
	at javax.swing.table.AbstractTableModel.fireTableDataChanged(Unknown Source)
	at mekhq.gui.model.DataTableModel.setData(DataTableModel.java:45)
	at mekhq.gui.RepairTab.refreshTaskList(RepairTab.java:798)
	at mekhq.gui.RepairTab.servicedUnitTableValueChanged(RepairTab.java:563)
	at mekhq.gui.RepairTab.lambda$initTab$3(RepairTab.java:252)
	at javax.swing.DefaultListSelectionModel.fireValueChanged(Unknown Source)
	at javax.swing.DefaultListSelectionModel.fireValueChanged(Unknown Source)
	at javax.swing.DefaultListSelectionModel.fireValueChanged(Unknown Source)
	at javax.swing.DefaultListSelectionModel.changeSelection(Unknown Source)
	at javax.swing.DefaultListSelectionModel.changeSelection(Unknown Source)
	at javax.swing.DefaultListSelectionModel.setSelectionInterval(Unknown Source)
	at javax.swing.JTable.changeSelectionModel(Unknown Source)
	at javax.swing.JTable.changeSelection(Unknown Source)
	at javax.swing.plaf.basic.BasicTableUI$Handler.adjustSelection(Unknown Source)
	at javax.swing.plaf.basic.BasicTableUI$Handler.mousePressed(Unknown Source)
	at java.awt.AWTEventMulticaster.mousePressed(Unknown Source)
	at java.awt.AWTEventMulticaster.mousePressed(Unknown Source)
	at java.awt.Component.processMouseEvent(Unknown Source)
	at javax.swing.JComponent.processMouseEvent(Unknown Source)
	at java.awt.Component.processEvent(Unknown Source)
	at java.awt.Container.processEvent(Unknown Source)
	at java.awt.Component.dispatchEventImpl(Unknown Source)
	at java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.awt.Component.dispatchEvent(Unknown Source)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)
	at java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)
	at java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)
	at java.awt.Container.dispatchEventImpl(Unknown Source)
	at java.awt.Window.dispatchEventImpl(Unknown Source)
	at java.awt.Component.dispatchEvent(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue$4.run(Unknown Source)
	at java.awt.EventQueue$4.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.run(Unknown Source)
```

### Files
[campaign save.zip](https://github.com/MegaMek/mekhq/files/2924313/campaign.save.zip)
[mechfiles.zip](https://github.com/MegaMek/mekhq/files/2924314/mechfiles.zip)
[mekhqlog.zip](https://github.com/MegaMek/mekhq/files/2924315/mekhqlog.zip)

**Comment by sixlettervariables** [comment]
I'm holding this open while I search for why your refit caused the mounted to no longer be correct.

**Comment by sixlettervariables** [comment]
I'm tentatively closing this as fixed in #1216. This will land in either 0.45.5 or 0.46.
